selftests/functional/test_requirements.py

```diff
@@ -0,0 +1,155 @@
+import os
+import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process, script
+from avocado.utils.network.ports import find_free_port
+from selftests.utils import AVOCADO, TestCaseTmpDir
+
+
+SINGLE_SUCCESS_CHECK = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class SuccessTest(Test):
+
+    def test_check(self):
+        """
+        :avocado: requirement={"type": "package", "name": "bash", "action": "check"}
+        """
+'''
+
+SINGLE_FAIL_CHECK = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class FailTest(Test):
+
+    def test_check(self):
+        """
+        :avocado: requirement={"type": "package", "name": "-foo-bar-", "action": "check"}
+        """
+'''
+
+MULTIPLE_SUCCESS = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class SuccessTest(Test):
+
+    def test_a(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_b(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_c(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+'''
+
+MULTIPLE_FAIL = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class FailTest(Test):
+
+    def test_a(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        :avocado: requirement={"type": "package", "name": "-foo-bar-"}
+        """
+    def test_b(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_c(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        :avocado: requirement={"type": "package", "name": "-foo-bar-"}
+        """
+'''
+
+
+class BasicTest(TestCaseTmpDir):
+
+    status_server = '127.0.0.1:%u' % find_free_port()
+    command = ('%s run --test-runner=nrunner'
+               ' --nrunner-status-server-uri %s'
+               ' --nrunner-status-server-listen %s %s')
+    skip_install_message = ("This test runs on CI environments only as it"
+                            " installs packages to test the feature, which"
+                            " may not be desired locally, in the user's"
+                            " system.")
+    skip_package_manager_message = ("This test runs on CI environments only"
+                                    " as it depends on the system package"
+                                    " manager, and some environments don't"
+                                    " have it available.")
+
+    @unittest.skipIf(not os.getenv('CI') and os.getenv('CI') not in ['true'],
```

Comment:
I need to confirm if os.getenv('CI') evaluates to `True` as it returns `true`; I think I tested that, and it didn't work; that is why I used the explicitly check. The leftover was the incorrect message.

Reply:
Right, but this also means that `CI=false` will evaluate to `True`. I know the likelihood of someone setting `CI=false` is low, but for consistency of evaluating a boolean variable, I would check for an explicitly `True`.

Reply:
> I need to confirm if os.getenv('CI') evaluates to `True` as it returns `true`; I think I tested that, and it didn't work; that is why I used the explicitly check. The leftover was the incorrect message.

I was having problems with `os.environ.get()`, not `os.getenv()`.

I'll wait on a final agreement about use an explicit check or not.

Reply:
Guys, I honestly feel in a bit like in twilight zone episode right now.  I see no reason why a CI system would **set** an environment variable like `CI=false`, let alone a system that is not a CI.  The equivalent to that is simply removing the variable, that is, `unset CI`.

And we have prior art of that working, like in the QEMU acceptance test, like here: https://gitlab.com/qemu-project/qemu/-/blob/master/tests/acceptance/boot_linux.py#L141

Reply:
Hi @clebergnu I understand your point, and probably we are ok with a non-explicit check on the value. I have changed my option and I'm more flexible here after thinking better about this problem.

However, let me explain my concerns and my comments:

> you're proposing a "CI variable standard" that...

No, I was not, I'm just trying to be extra safer and explicit here, thinking about our users, that might not be a CI expert.

> ... AFAICT, doesn't really exist in the community/industry. 

Is this convention of using `CI=yes` a really standard on the community/industry? Or just a few CI are using? I just don't like the idea that this is a very well defined standard. Jenkins for instance it was not using this `CI=true` env until a few weeks ago. Some CI systems might not having this set. So at the end, we need to do a extra step to create "our way". How Avocado will handle multiple CIs envs depend on us.

> Since you talked about "making assumptions", this will probably let people reading the code into assuming that this is something CI systems will do.

We are saying that this is how *our* code works: we are expecting this ENV set to true (or just set). It is not an assumption about third-party software that we don't control. If we use this same logic, "we are letting people assuming that *all* CI systems follows this standard (having a CI env).". 

Either your proposal or mine, will fail during this detection, in those systems that might not have the env, indeed. My only concern with the implicit check is this:

> unset CI is all that users should do if they want to disable the CI environment variable. Nothing else.

Users (for *any* reason, including lack of acknowledgement) could do `CI=false` and expect Avocado would respect this decision. Asking users to do `unset CI` to enable/disable extra tests IMO requires that users are aware of this small caveat and could create frustrations.

As I said, I'm OK with you guys decide, (I have already approved this PR) but IMO we are going to be user friendlier if we decide to be explicit here.

Reply:
> Now contrast this with `CI=false|no|0`. To the best of my knowledge, 0% of the CIs we know about do that.

Unfortunately, I already fall into the trap of setting `CI=false` to have a test run locally and I spent some time until I understand that it wouldn't work because the instruction was `skipIf(os.getenv ('CI'))`

Reply:
> > Now contrast this with `CI=false|no|0`. To the best of my knowledge, 0% of the CIs we know about do that.
> 
> Unfortunately, I already fall into the trap of setting `CI=false` to have a test run locally and I spent some time until I understand that it wouldn't work because the instruction was `skipIf(os.getenv ('CI'))`

Right... so IMO we should follow the de-facto standard we were able to observe in the CI systems "out there", **or** come up with a different variable to control the execution of these tests.  We have something like this with `AVOCADO_CHECK_LEVEL` btw.